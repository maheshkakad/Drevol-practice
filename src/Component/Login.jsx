import React, { useState } from "react";

const Login  = ({submit}) => {
    const [values, setValues] = useState({
        username: '',
        password: '',
    })

    const handleChange = (event) => {
        setValues({
          ...values,
          [event.target.id]: event.target.value,
        });
    }
    const handleSubmit = (event) => {
        event.preventDefault();
        submit(values)
    }
 console.log(values)
    return (
      <form onSubmit={handleSubmit}>

        <label htmlFor="username">Username</label>
        <input id="username" onChange={handleChange} value={values.username}/>

        <label htmlFor="password">Password</label>
        <input id="password" onChange={handleChange} value={values.password} />

        <button type="submit">Submit</button>

      </form>
    );
}

export default Login