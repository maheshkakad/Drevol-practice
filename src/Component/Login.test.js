import {render, screen} from "@testing-library/react"
import userEvent from '@testing-library/user-event'
import Login from "./Login";


it('should have a username and a password field, also a submit button', () => {
    render(<Login/>)

    const usernameField = screen.getByLabelText(/username/i);
    const passwordField = screen.getByLabelText(/password/i);
    const submitButton = screen.getByText(/submit/i);

    expect(usernameField).toBeInTheDocument();
    expect(passwordField).toBeInTheDocument();
    expect(submitButton).toBeInTheDocument();
});

it('should allow the user to submit their credentials', () => {
    
    const submit = jest.fn();
    render(<Login submit={submit}/>)

    const usernameField = screen.getByLabelText(/username/i);
    const passwordField = screen.getByLabelText(/password/i);
    const submitButton = screen.getByText(/submit/i);

    userEvent.type(usernameField, "remy");
    userEvent.type(passwordField, "vsecure");
    userEvent.click(submitButton);

    expect(submit).toHaveBeenCalledWith({
        username: "remy",
        password: "vsecure"
    });


})

// test("user name should be rendered" , ()=> {
//      render(<Login />)
//      const userInputEl = screen.getAllByPlaceholderText(/username/i)

//      expect(userInputEl).toBeInTheDocument()

// })

// test("password input should be rendered" , ()=> {
//     render(<Login />)
//     const passwordInputEl = screen.getAllByPlaceholderText(/password/i);

//     expect(passwordInputEl).toBeInTheDocument();
    
// })
// test(" button should be rendered" , ()=> {
//     render(<Login />)
//     const buttonInputEl = screen.getByRole("button");

//     expect(buttonInputEl).toBeInTheDocument();
    
// })

// test('render react title' , ()=> {
//   // let renderResult:RenderResult;
 
//    const component  = renderer.create( <Login />);

//    let tree = component.toJSON()

//    expect(tree).toMatchSnapshot();

// })