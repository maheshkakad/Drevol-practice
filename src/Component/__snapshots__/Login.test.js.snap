// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render react title 1`] = `
<div
  className="container"
>
  <form>
    <input
      placeholder="username"
      type="text"
    />
    <input
      placeholder="password"
      type="passsword"
    />
    <button>
      Login
    </button>
  </form>
</div>
`;
